scripts/qase-patch-validation.sh

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env bash
+
+# Qase Patch Validation Run Creation Script.
+
+PS4='+(${LINENO}): '
+set -e
+trap 'echo "Error on line $LINENO: $BASH_COMMAND"' ERR
+
+validate_token() {
+    if [[ -z "$QASE_API_TOKEN" ]]; then
+        echo "Error: Missing required var QASE_API_TOKEN."
+        exit 1
+    fi
+}
+
+set_vars() {
+    QASE_PROJECT_CODE='K3SRKE2'
+    QASE_TAG='team-rke2'
+    CURRENT_MONTH="$(date +"%B")"
+    CURRENT_YEAR="$(date +"%Y")"
+
+    QASE_MILESTONE="${CURRENT_MONTH}-${CURRENT_YEAR} Patch release"
+    echo "QASE_MILESTONE=$QASE_MILESTONE"
+
+    # Get the list of rcs to process from GH action parameter.
+    IFS=',' read -r -a rcs_to_process <<<"${RCS}"
+
+    All_RCS=${rcs_to_process[*]}
+}
+
+# Function to create milestone with given name. It will return milestone ID.
+create_milestone() {
+    if [[ -z "$QASE_MILESTONE" ]]; then
+        echo "Error: Missing required QASE_MILESTONE."
+        exit 1
+    fi
+
+    RESPONSE=$(curl --request POST \
+        --url "https://api.qase.io/v1/milestone/$QASE_PROJECT_CODE" \
+        --header "Token: $QASE_API_TOKEN" \
+        --header 'Content-Type: application/json' \
+        --data '{
+                "title": "'"$QASE_MILESTONE"'"
+            }')
+
+    # extract milestone ID from response.
+    MILESTONE_ID=$(echo "$RESPONSE" | jq '.result.id')
+    if [[ -z "$MILESTONE_ID" || "$MILESTONE_ID" == "null" ]]; then
+        echo "Failed to create milestone."
+        exit 1
+    fi
+
+    echo "$MILESTONE_ID"
+}
+
+process() {
+    read -r -a rcs <<<"${All_RCS}"
+    read -r -a products <<<"rke2 k3s"
+
+    versions=()
+    for rc  in "${rcs[@]}"; do
+      version="${rc%-rc*}"
+      versions+=("$version")
+    done
+
+    for product in "${products[@]}"; do
+        for ((i = 0; i < ${#versions[@]}; i++)); do
+
+            #we iterate by modulo to be safer.
+            VERSION="${versions[$((i % ${#versions[@]}))]}"
+            RC="${rcs[$((i % ${#rcs[@]}))]}"
+
+            if [ "$product" == "rke2" ]; then
+                QASE_TEST_PLAN_ID='14'
+                IDENTIFIER='rke2r1'
```

Comment:
What if we want to create plan for r2 / s2 🤞 ? It wont work right? Or we need/want to manually update it and have it to r1/s1 by default?

Reply:
this is not the plan..... the plan should not change as per s1,s2,s3 etc...

if we want to update in qase the description for s1 or s2 ...we will need to update manually, this is like a default

Reply:
this is creating the starter point, a way to create patch run automated, if in the middle of the process where its already created, we need to update manually